### PR TITLE
Start refresh while bouncing back to 0.0 from overscroll on iOS

### LINKED ISSUE #17059: RefreshIndicator pauses animation on iOS
## Steps to Reproduce

Open a Flutter app* with the pull-to-refresh interaction (`RefreshIndicator`) on iOS or in the iOS Simulator. Pull to refresh and watch the rotating arrow abruptly stop at one point.

*) For example, the Flutter Gallery app

refreshindicator mp4 gif
^^^ animated gif (takes time to load)

This does not happen on Android or in the Android Emulator.

## Flutter Doctor

```
$ flutter doctor -v
[✓] Flutter (Channel beta, v0.3.1, on Mac OS X 10.13.3 17D102, locale en-US)
    • Flutter version 0.3.1 at /Users/filiph/dev/flutter
    • Framework revision 12bbaba9ae (8 days ago), 2018-04-19 23:36:15 -0700
    • Engine revision 09d05a3891
    • Dart version 2.0.0-dev.48.0.flutter-fe606f890b

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/filiph/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.4.0

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 23.2.2
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 20.0.3
    • Dart plugin version 173.4127.31

[✓] VS Code (version 1.22.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Dart Code extension version 2.9.2

[!] Connected devices
    ! No devices available
```

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/17059.

### CODE REVIEW

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 1]
```diff
@@ -226,6 +226,9 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
           _checkDragOffset(notification.metrics.viewportDimension);
         }
       }
+      if (_mode == _RefreshIndicatorMode.armed && notification.dragDetails == null) {
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/refresh_indicator.dart]: It's not clear (to me anyway) that this particular line is intended to "Start refresh while bouncing back to 0.0 from overscroll on iOS". Why is this iOS specific and etc. Either I need to read a bunch more of the code, or some kind of comment should appear here.

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/refresh_indicator.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/material/refresh_indicator.dart]: Added comment.

File: packages/flutter/test/material/refresh_indicator_test.dart [hunk 2]
```diff
@@ -378,4 +379,41 @@ void main() {
...
+    expect(lastScrollOffset = controller.offset, lessThan(0.0));
+    expect(refreshCalled, isFalse);
+
+    await tester.pump(const Duration(milliseconds: 100));
+    expect(controller.offset, greaterThan(lastScrollOffset));
+    expect(controller.offset, lessThan(0.0));
+    expect(refreshCalled, isTrue);
+
+    debugDefaultTargetPlatformOverride = null;
+  });
```
Comment (Hixie) [inline packages/flutter/test/material/refresh_indicator_test.dart]: shouldn't we also check that it doesn't happen on android, if this is an iOS feature?

Comment (goderbauer) [reply packages/flutter/test/material/refresh_indicator_test.dart]: I believe that's covered by tests for Scrollables/ListViews to ensure that they don't overscroll on Android.

**Pull request merged**